app.rb

```diff
@@ -77,14 +111,86 @@ def self.valid?(username, password)
 
 helpers do
   def shared_secret
-    @shared_secret ||= File.read(FlightWebAuth.config.shared_secret_path)
+    FlightWebAuth.config.shared_secret
+  end
+
+  def sso_cookie_domain
+    if request.host == 'localhost'
+      'localhost'
+    elsif (request.host !~ /^[\d.]+$/) && (request.host =~ DOMAIN_REGEXP)
+      ".#{$&}"
+    end
+  end
+
+  def set_sso_cookie(auth_token)
+    response.set_cookie(
+      FlightWebAuth.app.config.sso_cookie_name,
+      domain: sso_cookie_domain,
```

Comment:
A possible way around the the regex issue described above is to force the `sso_cookie_domain` to be a configuration value. Or maybe a longest matching `case` statement.